avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
And how is this different to the `write_file`? The docstring should mention the differences. I'd recommend something like `Writes data to a file, logging the exception on failure`. for the current implementation, anyway ... (more is coming on the next lines)